ethcore/src/client/client.rs

```diff
@@ -909,6 +911,11 @@ impl Client {
 		*self.io_channel.lock() = io_channel;
 	}
 
+	/// Get io channel
+	pub fn get_io_channel(&self) -> Arc<Mutex<IoChannel<ClientIoMessage>>> {
```

Comment:
Exposing `Mutex`'s in public interface is a bad pattern. Use [channels](https://doc.rust-lang.org/std/sync/mpsc/fn.channel.html) instead